### DISCUSSION TITLE
The best way to add an additional datapoint without refitting the hyper-parameters

### DISCUSSION THREAD

**Discussion by azinoma** [GPflow/GPflow#1630] [Q&A]
I am wondering what would be the best way to add an additional data point to the GP, without re-optimizing it.
Currently I am doing something like:
```
(Xdata_old, Ydata_old) = model.data
Xdata_new = tf.concat([Xdata_old, Xdata_add], 0)
Ydata_new = tf.concat([Ydata_old, Ydata_add], 0)
model.data = (Xdata_new, Ydata_new)
```
This works, but I am wondering if there is a better or more proper way to do it.

Thanks a lot for your help.

**Comment by st--** [comment]
@azinoma this depends a bit more on your use-case - e.g. what model class are you using? In regression with Gaussian likelihood, in principle you can do a low-rank update of the posterior (but this isn't implemented in GPflow). In variational models, you can set the rows/columns of q_mu and q_sqrt so that adding new points doesn't change the posterior, and then optimize only the variational parameters. Again, this would require some custom code at the moment as far as I am aware.
